src/utilities.ts

```diff
@@ -19,6 +19,27 @@ export async function exec(command: string, options: cp.ExecOptions): Promise<{
     );
 }
 
+/**
+ * Extracts the base name of a repository from its URL.
+ * 
+ * The base name is the last path component of the URL, without the extension `.git`,
+ * and without an optional trailing slash.
+ */
+export function getRepositoryName(url: string): string {
+    // This regular expression consists of:
+    // - any number of characters that aren't a slash: ([^/]*)
+    // - optionally followed by a trailing slash: \/? 
+    // - at the end of the URL: $
+    const pattern = /([^/]*)\/?$/;
```

Comment:
Not a great fan of regular expressions. They are very opaque to users reading code. The comment above is very helpful

Reply:
Yeah same here. I like what I can do what them, but I don't like reading them, hence the comment :)